internal/services/pubkey_service.go

```diff
@@ -99,6 +109,51 @@ func DeletePubkey(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	pubkey, err := pubkeyDao.GetById(r.Context(), id)
+	if err != nil {
+		renderError(w, r, payloads.NewNotFoundError(r.Context(), err))
+		return
+	}
+
+	resources, err := pubkeyDao.UnscopedListByPubkeyId(r.Context(), pubkey.ID)
+	if err != nil {
+		renderError(w, r, payloads.NewDAOError(r.Context(), "delete pubkey", err))
+		return
+	}
+
+	for _, res := range resources {
+		if res.Handle != "" {
+			logger.Info().Msgf("Deleting pubkey resource ID %v with handle %s", res.ID, res.Handle)
+			arn, errArn := sourcesClient.GetAuthentication(r.Context(), res.SourceID)
+			if errArn != nil {
+				if errors.Is(err, httpClients.ApplicationNotFoundErr) {
+					renderError(w, r, payloads.ClientError(r.Context(), "Sources", "can't fetch arn from sources: application not found", errArn, 404))
+					return
+				}
+				if errors.Is(err, httpClients.AuthenticationForSourcesNotFoundErr) {
+					renderError(w, r, payloads.ClientError(r.Context(), "Sources", "can't fetch arn from sources: authentication not found", errArn, 404))
+					return
+				}
+				renderError(w, r, payloads.ClientError(r.Context(), "Sources", "can't fetch arn from sources", errArn, 500))
+				return
+			}
+
+			ec2Client, errEc2 := clients.GetCustomerEC2Client(r.Context(), arn, res.Region)
+			if errEc2 != nil {
+				renderError(w, r, payloads.NewAWSError(r.Context(), "failed to establish ec2 connection", errEc2))
+				return
+			}
+
+			errDelete := ec2Client.DeleteSSHKey(r.Context(), res.Handle)
+			if errDelete != nil {
+				renderError(w, r, payloads.NewAWSError(r.Context(), "can't delete public key", errDelete))
+				return
+			}
```

Comment:
The whole ec2 client deletion must be in a switch-case statement, see above.

Reply:
Doesn't have to be a switch statement, just make sure that the EC2 code is only executed when AWS authorization type is returned, otherwise raise an error. That's all.